### PR TITLE
Add install target, fix Python bindings & tests

### PR DESCRIPTION
**Pull request by xantares**
Automatically install Python bindings to the python site prefix and common libs relatively to CMAKE_INSTALL_PREFIX.
This is an alternative to setting PYTHON_OUTDIR_PREFIX and COMMON_LIBRARIES_PATH that allows to take rpath settings into account, also has the advantage to avoid to discard the CGAL*PYTHONwrap.cxx files.

Then various fixes related to buidling & testing in particular for Python 3.

I also propose a Travis config file that runs Python tests for Python2/3.

### CODE REVIEW

File: SWIG_CGAL/common.i [hunk 1]
```diff
@@ -279,7 +279,7 @@
 //exception for iterators (for next functions)
 #ifdef SWIGPYTHON
 %include exception.i
-%exception next
+%exception
```
Comment (sloriot) [inline SWIG_CGAL/common.i]: why this change?

Comment (xantares) [inline SWIG_CGAL/common.i]: For Python 3, this requires to apply to the __next__ method of iterators too.

**Comment by xantares** [comment]
@sloriot is this ok to merge ?

**Comment by xantares** [comment]
@sloriot I'm waiting on your input to integrate cgal this in conda-forge for the conda python distribution:
https://github.com/conda-forge/cgal-feedstock

File: SWIG_CGAL/common.i (hunk 1 above)
Comment (sloriot) [inline SWIG_CGAL/common.i]: In that case, I think the exception should only apply to __next__ and/or next to avoid pollution in the code generated by SWIG

**Comment by sloriot** [comment]
Having travis support is a good idea but here it is only done for python and not java.

File: SWIG_CGAL/common.i (hunk 1 above)
Comment (xantares) [inline SWIG_CGAL/common.i]: ok, done

File: SWIG_CGAL/common.i [hunk 2]
```diff
@@ -289,6 +289,17 @@
...
+%exception __next__
+{
+  try{
+      $action
+    }
+    catch(Stop_iteration){//TODO: throw a specify exception
+      SWIG_SetErrorObj(PyExc_StopIteration, SWIG_Py_Void());
+      SWIG_fail;
+    }
+}
```
Comment (sloriot) [inline SWIG_CGAL/common.i]: I'm wondering whether this is really needed since `__next()__` is calling `next()` that is already throwing.

Comment (xantares) [inline SWIG_CGAL/common.i]: __next()__ is not calling the wrapped version but the c++ internal version, so yes it's necessary

**Comment by xantares** [comment]
@sloriot added testing of java in travis

**Comment by sloriot** [comment]
I think I would have let example that you are saying are not working but that's not a big deal. I merged the PR. Thanks for the contribution.

**Comment by xantares** [comment]
i'll try to enable them all: https://github.com/CGAL/cgal-swig-bindings/pull/73.
Please enable Travis on the repo!
